### ISSUE TITLE
Extend evaluator's cache write timeout

### ISSUE THREAD

**Issue by josh-newman** [grailbio/reflow#77]
Right now workers may die a few minutes after the last exec finishes, but that might mean transfers to the cache may not complete (so some work is "lost").

There's a tradeoff here between paying for machine time to copy to the cache vs. saving time later because more things are cached, but it could be worthwhile.

**Comment by mariusae** [comment]
To be clear, this is [already done](
https://github.com/grailbio/reflow/blob/master/tool/run.go#L315): we should consider increasing the timeout.

**Comment by josh-newman** [comment]
Is there a separate timeout for total idleness (no execs and cache writes)?

That is, could workers wait indefinitely (or a few hours) for cache writes, and then an additional 10 mins of idle time before terminating? (Maybe this is exactly what you're suggesting?)

**Comment by mariusae** [comment]
I'm not sure how this would help: if it's idle, then waiting longer is not going to do anything at all (except cost more?)

**Comment by josh-newman** [comment]
Oh, I thought that was a user experience optimization so if you quickly rerun something you don't need to wait as long for cluster spinup. I'd be fine with setting total idle wait to 0.

**Comment by josh-newman** [comment]
To summarize, though, setting that 10 min time to be longer, configurable, or always infinite would be great.

**Comment by mariusae** [comment]
That's already managed as a [separate timeout](https://github.com/grailbio/reflow/blob/3b55affd0d0572e3685648beada7415c8baed797/reflowlet/reflowlet.go#L142) in the Reflowlet. A Reflowlet is considered idle only after uploads have finished.

**Comment by josh-newman** [comment]
Oh, I was reading incorrectly earlier. You're saying the reflow client is quitting after 10 mins? I'm assuming the reflowlets respond to that by terminating, rather than finishing their upload work?

**Comment by mariusae** [comment]
The termination sequence is this:

- evaluation completes; the result is ready
- the evaluator waits for 10m for cache uploads to finish
- the worker becomes idle
- the worker waits for 10m for new work, or else shuts down

**Comment by josh-newman** [comment]
I see, thanks for clarifying.
